Firestore/Source/API/FIRFirestore+Internal.h

```diff
@@ -54,7 +55,7 @@ NS_ASSUME_NONNULL_BEGIN
 - (void)shutdownWithCompletion:(nullable void (^)(NSError *_Nullable error))completion
     NS_SWIFT_NAME(shutdown(completion:));
 
-@property(nonatomic, strong, readonly) FSTDatabaseID *databaseID;
+@property(nonatomic, assign, readonly) firebase::firestore::model::DatabaseId databaseID;
```

Comment:
One major difference between this and the Objective-C version that came before is that `FSTDatabaseID *` is extremely cheap to return here and so the Objective-C property that just returns it makes sense.

This property now makes a copy of `DatabaseId` which seems wrong. Likely what we want is to return `const DatabaseId&` but this means we can't use an Objective-C property declaration.

For DatabaseId this doesn't matter that much, but for some of the larger objects this will be a problem, especially for things that must be singletons within the system.

For readonly properties we can probably make the type of the property `const Foo&` and then manually declare the corresponding instance variable.

Alternatively, we could just decide that this object is owned by FSTFirestoreClient and continue to return this as `DatabaseId*`.